flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java

```diff
@@ -487,23 +487,22 @@ public ResultFetcher callStopJobOperation(
                                                         .get(
                                                                 clientTimeout.toMillis(),
                                                                 TimeUnit.MILLISECONDS));
-                                    } catch (Exception e) {
-                                        throw new FlinkException(
-                                                "Could not stop job "
-                                                        + stopJobOperation.getJobId()
-                                                        + " in session "
-                                                        + handle.getIdentifier()
-                                                        + ".",
-                                                e);
+                                    } else {
+                                        clusterClient.cancel(JobID.fromHexString(jobId)).get();
+                                        return Optional.empty();
                                     }
-                                } else {
-                                    clusterClient.cancel(JobID.fromHexString(jobId));
-                                    return Optional.empty();
+                                } catch (Exception e) {
+                                    throw new FlinkException(e);
```

Comment:
Can we directly throw SqlExecutionExcetpion here? We can also catch FlinkException in the `runClusterAction`.

Reply:
Good point. Fixed in https://github.com/apache/flink/pull/21582.